### PR TITLE
Added VTOL Takeoff navigation mode

### PR DESCRIPTION
**Pull request by RomanBapst**
**Describe problem solved by this pull request**
Please have a look at the corresponding pull request on QGroundControl:
https://github.com/mavlink/qgroundcontrol/pull/10123

### CODE REVIEW

File: src/lib/events/enums.json [hunk 1]
```diff
@@ -239,6 +239,10 @@
                             "name": "orbit",
                             "description": "Orbit"
                         },
+                        "15": {
+                            "name": "auto_vtol_takeoff",
+                            "description": "Vtol takeoff"
```
Comment (sfuhrer) [inline src/lib/events/enums.json:244]:
```suggestion
                            "description": "VTOL Takeoff"
```

File: src/modules/navigator/navigator.h [hunk 2]
```diff
@@ -78,14 +79,15 @@
 #include <uORB/topics/vehicle_land_detected.h>
 #include <uORB/topics/vehicle_local_position.h>
 #include <uORB/topics/vehicle_status.h>
+#include <uORB/topics/wind.h>
```
Comment (sfuhrer) [inline src/modules/navigator/navigator.h:82]: don't think wind is needed for the VTOL takeoff, is it?

File: src/modules/navigator/navigator_main.cpp [hunk 3]
```diff
@@ -455,6 +458,16 @@ Navigator::run()
...
 
+			} else if (cmd.command == vehicle_command_s::VEHICLE_CMD_NAV_VTOL_TAKEOFF) {
+
+				_vtol_takeoff.setTransitionAltitudeAbsolute(cmd.param7);
+
+				// after the transition the vehicle will establish on a loiter at this position
+				_vtol_takeoff.setLoiterLocation(matrix::Vector2d(cmd.param5, cmd.param6));
+
+				// loiter height is the height above takeoff altitude at which the vehicle will establish on a loiter circle
+				_vtol_takeoff.setLoiterHeight(cmd.param1);
```
Comment (sfuhrer) [inline src/modules/navigator/navigator_main.cpp:469]: That's not aligned with current mavlink specs right? Do you plan to open a mavlink PR for it, and meanwhile use VTO_LOITER_ALT instead?

File: src/modules/navigator/vtol_takeoff.cpp [hunk 4]
```diff
@@ -0,0 +1,178 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```
Comment (sfuhrer) [inline src/modules/navigator/vtol_takeoff.cpp:3]: 2022

File: src/modules/navigator/vtol_takeoff.cpp [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+				_takeoff_state = vtol_takeoff_state::TRANSITION;
+
+				break;
+			}
+
+		case vtol_takeoff_state::TRANSITION: {
+				position_setpoint_triplet_s *pos_sp_triplet = _navigator->get_position_setpoint_triplet();
+
+				if (pos_sp_triplet->current.valid && pos_sp_triplet->current.type == position_setpoint_s::SETPOINT_TYPE_LOITER) {
```
Comment (sfuhrer) [inline src/modules/navigator/vtol_takeoff.cpp:103]: In which case is that condition valid (type LOITER) and when the other?

File: src/modules/navigator/vtol_takeoff.cpp [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+				mission_item_to_position_setpoint(_mission_item, &pos_sp_triplet->current);
+				pos_sp_triplet->current.lat = _loiter_location(0);
+				pos_sp_triplet->current.lon = _loiter_location(1);
+				pos_sp_triplet->current.type = position_setpoint_s::SETPOINT_TYPE_LOITER;
+
+				_mission_item.lat = pos_sp_triplet->current.lat;
+				_mission_item.lon = pos_sp_triplet->current.lon;
+
+
+				//publish_navigator_mission_item(); // for logging
```
Comment (sfuhrer) [inline src/modules/navigator/vtol_takeoff.cpp:128]: do we need publish_navigator_mission_item() or not?

**Comment by RomanBapst** [comment]
@sfuhrer Thanks for the review, will address shortly

**Comment by RomanBapst** [comment]
@dagar Would you mind having a quick look at this as well?

File: src/modules/navigator/navigator_main.cpp (hunk 3 above)
Comment (RomanBapst) [reply src/modules/navigator/navigator_main.cpp:461]: @sfuhrer Yes, exactly. The value is currently not used, we still stick to the parameter. In order to use the value from the command we need to extend the mavlink specs and also add some UI in QGC for specifying the altitude.

**Comment by RomanBapst** [comment]
@sfuhrer I've addressed your comments which also lead to more cleanup in the vtol takeoff navigation mode. We had some residual stuff in there that was not needed according to my testing.

**Review by sfuhrer — APPROVED** [review]
All my comments were addressed and SITL testing passed.

**Pull request merged**